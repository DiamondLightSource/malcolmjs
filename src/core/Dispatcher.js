/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
