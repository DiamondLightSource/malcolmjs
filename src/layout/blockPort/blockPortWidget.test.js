import React from 'react';
import configureStore from 'redux-mock-store';
import { createShallow } from '@material-ui/core/test-utils';
import BlockPortWidget from './blockPortWidget.component';

describe('BlockPortWidget', () => {
  let shallow;
  let mockStore;
  let port;
  let state;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
    mockStore = configureStore();

    port = {
      name: 'val1',
      in: true,
      label: 'val 1',
      getParent: () => ({
        getID: () => '123456',
      }),
    };

    state = {
      malcolm: {
        layoutEngine: {
          diagramModel: {
            nodes: {
              node1: {
                ports: {
                  val1: port,
                },
              },
            },
          },
        },
      },
    };
  });

  it('input port renders correctly', () => {
    port.in = true;

    const wrapper = shallow(
      <BlockPortWidget nodeId="node1" portId="val1" store={mockStore(state)} />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('output port renders correctly', () => {
    port.in = false;

    const wrapper = shallow(
      <BlockPortWidget nodeId="node1" portId="val1" store={mockStore(state)} />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
