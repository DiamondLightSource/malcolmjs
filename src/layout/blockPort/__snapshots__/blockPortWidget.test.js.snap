// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockPortWidget input port renders correctly 1`] = `
<div
  className="BlockPortWidget-container-1"
>
  <div
    className="port BlockPortWidget-port-2 malcolm-port "
    data-name="val1"
    data-nodeid="node1"
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    role="presentation"
    style={
      Object {
        "background": "blue",
        "marginRight": 3,
      }
    }
  />
  <WithStyles(Typography)
    className="BlockPortWidget-portLabel-5"
  >
    val 1
  </WithStyles(Typography)>
</div>
`;

exports[`BlockPortWidget input port renders correctly with hidden link 1`] = `
<div
  className="BlockPortWidget-container-1"
>
  <div
    className="BlockPortWidget-hiddenLink-6"
  >
    <WithStyles(Typography)>
      HIDDEN_BLOCK.VAL
    </WithStyles(Typography)>
  </div>
  <div
    className="port BlockPortWidget-port-2 malcolm-port "
    data-name="val1"
    data-nodeid="node1"
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    role="presentation"
    style={
      Object {
        "background": "blue",
        "marginRight": 3,
      }
    }
  />
  <WithStyles(Typography)
    className="BlockPortWidget-portLabel-5"
  >
    val 1
  </WithStyles(Typography)>
</div>
`;

exports[`BlockPortWidget output port renders correctly 1`] = `
<div
  className="BlockPortWidget-container-1"
>
  <WithStyles(Typography)
    className="BlockPortWidget-portLabel-5"
  >
    val 1
  </WithStyles(Typography)>
  <div
    className="port BlockPortWidget-port-2 malcolm-port "
    data-name="val1"
    data-nodeid="node1"
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    role="presentation"
    style={
      Object {
        "background": "blue",
        "marginLeft": 3,
      }
    }
  />
</div>
`;

exports[`BlockPortWidget output port renders correctly with hidden link 1`] = `
<div
  className="BlockPortWidget-container-1"
>
  <WithStyles(Typography)
    className="BlockPortWidget-portLabel-5"
  >
    val 1
  </WithStyles(Typography)>
  <div
    className="port BlockPortWidget-port-2 malcolm-port "
    data-name="val1"
    data-nodeid="node1"
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    role="presentation"
    style={
      Object {
        "background": "blue",
        "marginLeft": 3,
      }
    }
  />
  <div
    className="BlockPortWidget-hiddenLink-6"
  >
    <WithStyles(Typography)>
      HIDDEN_BLOCK.VAL
    </WithStyles(Typography)>
  </div>
</div>
`;
